packages/block-editor/src/components/list-view/index.js

```diff
@@ -221,6 +227,8 @@ function ListViewComponent(
 			BlockSettingsMenu,
 			instanceId,
 			renderAdditionalBlockUI,
+			insertedBlock,
+			setInsertedBlock,
```

Comment:
It's always a potential problem for performance because if the context changes and components which consume it (or their children) will be re-rendered.

The setter is stable, but the `insertedBlock` will change. 

However, pre-optimising context is challenging. Better to include in the `memo` and then optimise if/when List View becomes a problem.